packages/trees/frontend/sirius-components-trees/src/treeitems/__tests__/filterTreeItem.test.ts

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { expect, test } from 'vitest';
+import { splitText } from '../filterTreeItem';
+
+test('should split text in a case insensitive manner', () => {
```

Comment:
Strange that this does not test for splitting on the special characters which triggers this whole ticket.